packages/app/src/actions/index.js

```diff
@@ -187,9 +188,22 @@ export const tDoSaveVisualization = ({ name, description }, copy) => async (
         if (copy) {
             delete visualization.id
         }
-
         if (name) {
             visualization.name = name
+        } else {
+            visualization.name = i18n.t(
+                'Untitled {{visualizationType}} visualization, {{date}}',
+                {
+                    visualizationType: getDisplayNameByVisType(
+                        visualization.type
+                    ),
+                    date: new Date().toLocaleDateString(undefined, {
+                        year: 'numeric',
+                        month: 'short',
+                        day: '2-digit',
+                    }),
+                }
+            )
         }
```

Comment:
Maybe just do `visualization.name = name || .....`, the if/else is not necessary when `name` is being set in both cases.

Reply:
Fair enough, I was just following the if-pattern of the rest of the file. Updated 👍